qutebrowser/browser/shared.py

```diff
@@ -329,10 +330,21 @@ def feature_permission(url, option, msg, yes_action, no_action, abort_on,
                 cancel_action=no_action, abort_on=abort_on,
                 title='Permission request', text=text, url=urlstr,
                 option=option)
-    elif config_val:
+
+    if isinstance(opt.typ, configtypes.AsBool):
+        config_val = opt.typ.to_bool(config_val)
+
+    if config_val is True:
         yes_action()
         return None
+    elif config_val is False:
+        no_action()
+        return None
     else:
+        log.misc.warning(
+            f"Unsupported value for permission prompt setting ({option}), expected boolean or "
+            f"'ask', got: {config_val} ({type(config_val)})"
+        )
```

Comment:
This branch gets entered if the value of a setting linked to a feature is anything other than True, False or "ask". I think this could only happen due to a programming error, for example when you add an entry to `_WebEnginePermissions` that was linked to a String type setting. So I think putting an assert here instead of a warning should be fine and more explicit. (Or should be be `utils.Unreachable`? Or `ValueError`?)

I'm not sure about making `Bool` implement `AsBool`. It might save a condition here but it seems a bit odd. I'll have a think about it, but an leaning toward waiting until a stronger reason comes along for that.